### PR TITLE
fix(sortable-list): updated to use column layout

### PR DESCRIPTION
**Pull request by asangma**
**Related Issue:** #2889

## Summary
* Updates to use flex column layout 
* Adds `layout` property that defaults to "vertical"

### CODE REVIEW

File: src/components/calcite-sortable-list/calcite-sortable-list.tsx [hunk 1]
```diff
@@ -200,6 +207,19 @@ export class CalciteSortableList {
   // --------------------------------------------------------------------------
 
   render(): VNode {
-    return <slot />;
+    const { layout } = this;
+    const layoutProp = layout == "vertical" || layout == "horizontal" ? layout : "vertical";
+    const layoutClass = layoutProp === "vertical" ? CSS.containerVertical : CSS.containerHorizontal;
```
Comment (asangma) [inline src/components/calcite-sortable-list/calcite-sortable-list.tsx:212]: @driskull If you have a suggestion for a cleaner way to do this, I'd love to know it. :)

Comment (driskull) [reply src/components/calcite-sortable-list/calcite-sortable-list.tsx:212]: @asangma how about this?

```ts
const layoutClass =(layout ?? "vertical") === "vertical" ? CSS.containerVertical : CSS.containerHorizontal;
```

or 

```ts
const layoutProp = layout  ?? "vertical";
```

```ts
class={{
          [CSS.container]: true,
          [CSS.containerVertical]: layoutProp === "vertical",
          [CSS. containerHorizontal]: layoutProp === "horizontal"
        }}
```

**Review by driskull — CHANGES_REQUESTED** [review]
Just some code cleanup

File: src/components/calcite-sortable-list/calcite-sortable-list.tsx (hunk 1 above)
Comment (asangma) [reply src/components/calcite-sortable-list/calcite-sortable-list.tsx:212]: Ah...that'd work if checking for a null value yeah?
I think our default value should handle that.

I'm concerned about an invalid value, e.g. "foo".

I guess I could redo the style to be column and add a row rule when the horizontal CSS class is added?

Comment (driskull) [reply src/components/calcite-sortable-list/calcite-sortable-list.tsx:212]: Yeah that would be better. make the fallback a little easier.

```ts
class={{
          [CSS.container]: true,
          [CSS.containerVertical]: effectiveLayout !== "horizontal",
          [CSS. containerHorizontal]: effectiveLayout === "horizontal"
        }}
```

Comment (driskull) [reply src/components/calcite-sortable-list/calcite-sortable-list.tsx:212]:
```
const horizontal = layout === "horizontal"  ||  false;

class={{
          [CSS.container]: true,
          [CSS.containerVertical]: !horizontal,
          [CSS. containerHorizontal]: horizontal
        }}
```

Comment (asangma) [reply src/components/calcite-sortable-list/calcite-sortable-list.tsx:212]: Thanks @driskull.

**Comment by asangma** [comment]
Thanks @driskull. Your suggestion is kleen.

File: src/components/calcite-sortable-list/calcite-sortable-list.tsx [hunk 2]
```diff
@@ -44,6 +46,11 @@ export class CalciteSortableList {
    */
   @Prop() handleSelector = "calcite-handle";
 
+  /**
+   * Indicates the horizontal or vertical orientation of the component.
+   */
+  @Prop({ reflect: true }) layout: Layout = "vertical";
```
Comment (driskull) [inline src/components/calcite-sortable-list/calcite-sortable-list.tsx:52]: Does it need to be reflected? Seems like it doesn't.

**Review by driskull — APPROVED** [review]
👍

File: src/components/calcite-sortable-list/calcite-sortable-list.tsx (hunk 2 above)
Comment (asangma) [reply src/components/calcite-sortable-list/calcite-sortable-list.tsx:52]: Makes sense. Will update.

**Pull request merged**